### PR TITLE
Stevenic/combine context

### PR DESCRIPTION
**Pull request by Stevenic**

### CODE REVIEW

File: libraries/botbuilder-dialogs/src/prompts/activityPrompt.ts [hunk 1]
```diff
@@ -45,22 +45,16 @@ export abstract class ActivityPrompt extends Dialog {
         const recognized: PromptRecognizerResult<Activity> = await this.onRecognize(dc.context, state.state, state.options);
 
         // Validate the return value
-        let end: boolean = false;
-        let endResult: any;
-        await this.validator(dc.context, {
+        let isValid = await this.validator({
```
Comment (benbrown) [inline libraries/botbuilder-dialogs/src/prompts/activityPrompt.ts]: could be a const

File: libraries/botbuilder-dialogs/src/prompts/activityPrompt.ts [hunk 2]
```diff
@@ -45,22 +45,16 @@ export abstract class ActivityPrompt extends Dialog {
         const recognized: PromptRecognizerResult<Activity> = await this.onRecognize(dc.context, state.state, state.options);
```
[the remark this answers is not in the document]
Comment (Stevenic) [reply libraries/botbuilder-dialogs/src/prompts/activityPrompt.ts]: The ActivityPrompt **requires** you pass in a validator (its not optional) so we don't need the extra check. I had to double check that myself when going through and updating the code but good eye.

File: libraries/botbuilder-dialogs/src/prompts/activityPrompt.ts (hunk 1 above)
Comment (Stevenic) [reply libraries/botbuilder-dialogs/src/prompts/activityPrompt.ts]: totally... I'll update that

File: libraries/botbuilder-dialogs/src/prompts/activityPrompt.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (Stevenic) [reply libraries/botbuilder-dialogs/src/prompts/activityPrompt.ts]: I added a comment to better call that out

**Comment by benbrown** [comment]
Looks good! I think the single StepContext will make lives easier!

File: libraries/botbuilder-dialogs/tests/attachmentPrompt.test.js [hunk 3]
```diff
@@ -57,10 +57,9 @@ describe('AttachmentPrompt', function() {
 
         const dialogState = convoState.createProperty('dialogState');
         const dialogs = new DialogSet(dialogState);
-        dialogs.add(new AttachmentPrompt('prompt', (context, prompt) => {
-            assert(context);
+        dialogs.add(new AttachmentPrompt('prompt', async (prompt) => {
             assert(prompt);
-            prompt.end(prompt.recognized.value);
+            return prompt.recognized.succeeded;
```
Comment (stevengum) [inline libraries/botbuilder-dialogs/tests/attachmentPrompt.test.js]: @v-ducvo, @kaiqb, here's a delta in how to use validators that needs to be documented.

**Pull request merged**